etherman/etherman.go

```diff
@@ -861,6 +849,160 @@ func (etherMan *Client) getGasPrice(ctx context.Context) *big.Int {
+	}
+	signedTx, err := auth.Signer(auth.From, tx)
+	if err != nil {
+		return nil, err
+	}
+	return signedTx, nil
+}
+
+// GetRevertMessage tries to get a revert message of a transaction
+func (etherMan *Client) GetRevertMessage(ctx context.Context, tx *types.Transaction) (string, error) {
+	if tx == nil {
+		return "", nil
+	}
+
+	receipt, err := etherMan.GetTxReceipt(ctx, tx.Hash())
+	if err != nil {
+		return "", err
+	}
+
+	if receipt.Status == types.ReceiptStatusFailed {
+		revertMessage, err := operations.RevertReason(ctx, etherMan.EthClient, tx, receipt.BlockNumber)
+		if err != nil {
+			return "", err
+		}
+		return revertMessage, nil
+	}
+	return "", nil
+}
+
+// AddOrReplaceAuth adds an authorization or replace an existent one to the same account
+func (etherMan *Client) AddOrReplaceAuth(auth bind.TransactOpts) error {
+	log.Infof("added or replaced authorization for address: %v", auth.From.String())
+	etherMan.auth[auth.From] = auth
+	return nil
+}
+
+// LoadAuthFromKeyStore loads an authorization from a key store file
+func (etherMan *Client) LoadAuthFromKeyStore(path, password string) (*bind.TransactOpts, error) {
+	auth, err := newAuthFromKeystore(path, password, etherMan.cfg.L1ChainID)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("loaded authorization for address: %v", auth.From.String())
+	etherMan.auth[auth.From] = auth
+	return &auth, nil
+}
+
+// newKeyFromKeystore creates an instance of a keystore key from a keystore file
+func newKeyFromKeystore(path, password string) (*keystore.Key, error) {
+	if path == "" && password == "" {
+		return nil, nil
+	}
+	keystoreEncrypted, err := ioutil.ReadFile(filepath.Clean(path))
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("decrypting key from: %v", path)
+	key, err := keystore.DecryptKey(keystoreEncrypted, password)
+	if err != nil {
+		return nil, err
+	}
+	return key, nil
+}
+
+// newAuthFromKeystore an authorization instance from a keystore file
+func newAuthFromKeystore(path, password string, chainID uint64) (bind.TransactOpts, error) {
+	log.Infof("reading key from: %v", path)
+	key, err := newKeyFromKeystore(path, password)
+	if err != nil {
+		return bind.TransactOpts{}, err
+	}
+	if key == nil {
+		return bind.TransactOpts{}, nil
+	}
+	auth, err := bind.NewKeyedTransactorWithChainID(key.PrivateKey, new(big.Int).SetUint64(chainID))
+	if err != nil {
+		return bind.TransactOpts{}, err
+	}
+	return *auth, nil
+}
+
+// getAuthByAddress tries to get an authorization from the authorizations map
+func (etherMan *Client) getAuthByAddress(addr common.Address) (bind.TransactOpts, error) {
+	auth, found := etherMan.auth[addr]
+	if !found {
+		return bind.TransactOpts{}, ErrNotFound
+	}
+	return auth, nil
+}
+
+// generateRandomAuth generates an authorization instance from a
+// randomly generated private key to be used to estimate gas for PoE
+// operations NOT restricted to the Trusted Sequencer
+func (etherMan *Client) generateRandomAuth() (bind.TransactOpts, error) {
```

Comment:
Well, in this new version of the `EthTxManager`, the `Sequencer` and `Aggregator` shouldn't know about the account secrets and only about the public address, the `EthTxManager` is the holder of the secrets and is the only one able to sign the Txs with it. Also, all the `Transaction` logic happens inside of the `EthTxManager` and they are not triggered by the `Sequencer` or `Aggregator` using `Etherman`.

With that said, `Etherman` should be used by the `Sequencer` and `Aggregator` in read-only mode and shouldn't require an authorization object, BUT the generated code to interact with the SC doesn't help much in terms of creating a Tx without signing it, because the methods are meant to send the TX to the blockchain, so in order to use these methods we must provide an authorization object.

That's why we have this method.